packages/ag-charts-community/src/api/state/initialStateModule.ts

```diff
@@ -0,0 +1,11 @@
+import type { Module } from '../../module/module';
+import { InitialState } from './initialState';
+
+export const InitialStateModule: Module = {
+    type: 'root',
+    optionsKey: 'initialState',
+    dependencies: ['annotations'],
```

Comment:
This depends on annotations, which in turn depends on toolbar. The previous depedency ordering logic could not handle this, so required the change in this pr.

Reply:
I'm wondering if this should be the other way around though - it doesn't make sense for a community feature to depend on an enterprise feature?

Could we work the other way around, and have the `InitialStateModule` a dependency of `AnnotationModule` - and the later registers with the former at runtime to percolate `get`/`set` handling for the registered sub-tree?

If a user is only using `ag-charts-community`, then we would only support the null-set of root properties right no? (The typings would still reflect the enterprise super-set though)?

Reply:
Added `StateManager` onto the module context. On `initialState.annotations` change it updates the state on the manager. Then `Annotations.constructor()` calls `StateManager.requestRestore()` to trigger a restoration if there is any state available.